DFC.Api.Content/Function/Execute.cs

```diff
@@ -0,0 +1,142 @@
+using DFC.ServiceTaxonomy.ApiFunction.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Web.Http;
+using DFC.ServiceTaxonomy.ApiFunction.Exceptions;
+using ILogger = Microsoft.Extensions.Logging.ILogger;
+using DFC.ServiceTaxonomy.Neo4j.Services;
+using DFC.Api.Content.Models.Cypher;
+using Neo4j.Driver;
+using System.Linq;
+using Newtonsoft.Json;
+using DFC.Api.Content.Helpers;
+using DFC.Api.Content.Enums;
+using DFC.Api.Content.Models;
+
+namespace DFC.ServiceTaxonomy.ApiFunction.Function
+{
+    public class Execute
+    {
+        private readonly IOptionsMonitor<ContentTypeMapSettings> _contentTypeMapSettings;
+        private readonly IGraphDatabase _graphDatabase;
+        private readonly IJsonFormatHelper _jsonFormatHelper;
+        private const string contentByIdCypher = "MATCH (s {{uri:'{0}'}}) optional match(s)-[r]->(d) with s, {{href:d.uri, type:'GET', title:d.skos__prefLabel, dynamicKey:reduce(lab = '', n IN labels(d) | case n WHEN 'Resource' THEN '' ELSE n END), rel:labels(d)}} as destinationUris with s, apoc.map.fromValues([destinationUris.dynamicKey, {{href: destinationUris.href }}]) as map with s, collect(map) as links with s,links,{{ data: properties(s)}} as sourceNodeWithOutgoingRelationships return {{data:sourceNodeWithOutgoingRelationships.data, _links:links}}";
+
+        private const string contentGetAllCypher = "MATCH (n:{0}) with {{properties: properties(n)}} as data return data.properties;";
+
+        public Execute(IOptionsMonitor<ContentTypeMapSettings> contentTypeMapSettings, IGraphDatabase graphDatabase, IJsonFormatHelper jsonFormatHelper)
+        {
+            _contentTypeMapSettings = contentTypeMapSettings ?? throw new ArgumentNullException(nameof(contentTypeMapSettings));
+            _graphDatabase = graphDatabase ?? throw new ArgumentNullException(nameof(graphDatabase));
+            _jsonFormatHelper = jsonFormatHelper ?? throw new ArgumentNullException(nameof(jsonFormatHelper));
+        }
+
+        [FunctionName("Execute")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Execute/{contentType}/{id:guid?}")] HttpRequest req, string contentType, Guid? id,
+            ILogger log)
+        {
+            try
+            {
+                SetContentTypeHeader(req);
```

Comment:
Perhaps we should move this to just before the return, in case it adds it to error responses (where the response isn't in hal+json format).